Adds decorate functionality to module output (#1357)
insightsengineering/teal: vignettes/data-transform-as-shiny-module.Rmd

## Introduction

> `teal` version `0.16` introduced new argument in `teal::module` called `transformators`. 

Reviewer: ```suggestion
`teal` version `0.16` introduced a new, optional argument in `teal::module` named `transformators`. 
```

> This argument allows to pass a `list` of `teal_data_module` objects that are created using `teal_transform_module()` function.

Reviewer: ```suggestion
This argument allows to pass a `list` of `"teal_data_module"` class of objects created using [`teal_transform_module()`](https://insightsengineering.github.io/teal/main/reference/teal_transform_module.html) function.
```

Reviewer: - put class name as a string
- if you refer to function - link it - it would be easier to navigate through the docs

> This argument allows to pass a `list` of `teal_data_module` objects that are created using `teal_transform_module()` function.

Reviewer: There is a statement in line 21 "When provided, ...". Move it here. Start with "This is an optional argument and if it is provided then ..."

Here we are discussing the argument and that statement is indeed an argument description

Reviewer: Oh no - scratch that. We are describing the benefits of `transformators` argument - not the benefits of a function. That needs to be changed accordingly. Also probably should be moved upper so that we have: in new version we have added X. X is good because of .... In order to create X use foo_x(). foo_x() is ...

> The main benefit of `teal_transform_module()` is the ability to transform data before passing it 

Reviewer: There is a statement in line 21: "teal_transform_module()` is a function ...". Move it here - upgfront. Let's introduce the function first and then discuss it's benefits.

Reviewer 2: 29ea4f17cc7008c4b9731fd6889b2fd8847cc067

to the module. This feature allows to extend the regular behavior of existing modules by specifying custom data operations on data inside this module.

> `teal_transform_module()` is a Shiny module that takes `ui` and `server` arguments. When provided, `teal` will execute data transformations for the specified module when it is loaded and whenever the data changes. `server` extend the logic behind data manipulations, where `ui` extends filter panel with new UI elements that orchestrate the transformator inputs.

Reviewer 2: 1bdd1592c7b991e1c8bdfb4437a1901e82f7bd21
